services/web-server/src/login/strategies/mozilla-auth0.js

```diff
@@ -88,15 +87,23 @@ module.exports = class MozillaAuth0 {
     return user;
   }
 
-  userFromIdentity(identity) {
+  async userFromIdentity(identity) {
     let encodedUserId = identity.split('/')[1];
 
     if (encodedUserId.startsWith('github|') || encodedUserId.startsWith('oauth2|firefoxaccounts|')) {
       encodedUserId = encodedUserId.replace(/\|[^|]*$/, '');
     }
 
     const userId = decode(encodedUserId);
-    return this.getUser({ userId });
+    const user = await this.getUser({ userId });
+
+    // catch cases where the calculated identity differs, such as when we add a
+    // GitHub username, and return no user in that case.
```

Comment:
I don't particularly understand what this means.

Reply:
See https://github.com/taskcluster/taskcluster/blob/8be8f81d521d514548b1225fcd59babdeb3c5be6/services/web-server/docs/login-strategies.md#identities

Reply:
yeah but is `userId` not the numeric id? Why do we return no user at all?

I probably just need to read this more in depth.

Reply:
It is.  The decision here is that if you changed your username to `bstack`, your identity would go from `mozilla-auth0/github|127521|imbstack` to `mozilla-auth0/github|127521|bstack`, and that's a different identity.  So at that point `mozilla-auth0/github|127521|imbstack` is no longer a valid identity, and we return no user for it.